TpmTestingPkg/TpmReplayPeiDxe/TpmReplayReportingManager.c

```diff
@@ -0,0 +1,43 @@
+/** @file
+  TPM Replay Reporting Manager - Manages reporting of TPM Replay status.
+
+  Copyright (c) Microsoft Corporation.
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+
+#include "TpmReplayReportingManager.h"
+
+#include <Uefi.h>
+#include <Library/DebugLib.h>
+
+/**
+  Reports a TPM replay error.
+
+  A common reporting implementation for the feature that can be updated to report to sources
+  such as serial, status code reporting, telemetry, etc.
+
+  @param[in]  Error             The error code.
+  @param[in]  FunctionName      The function name the error occurred in.
+
+**/
+VOID
+ReportTpmReplayError (
+  IN  TPM_REPLAY_ERROR  Error,
+  IN  CONST CHAR8       *FunctionName   OPTIONAL
+  )
+{
+  CONST CHAR8  *DebugFunctionName;

```

Comment:
I'm surprised this isn't a build break since you change the value of a CONST?

Reply:
This is a pointer to constant data (not a constant pointer). The pointer can be moved (as is done here) but the data it points to cannot be manipulated (it is not here).